apps/navstar_dns/src/navstar_dns_key_mgr.erl

```diff
@@ -230,6 +234,23 @@ do_create_zk_key(Pid) ->
             false
     end.
 
+encode(#{public := Pk, secret := Sk}) ->
+    {Encoding, 0} = unicode:bom_to_encoding(Pk),
+    PkUtf8 = change_encoding(Pk, Encoding, utf8),
+    SkUtf8 = change_encoding(Sk, Encoding, utf8),
+    jsx:encode(#{public => PkUtf8, secret => SkUtf8, encoding => Encoding}).
+
+decode(Data) ->
+    #{public := PkUtf8, secret := SkUtf8, encoding := EncodingUtf8} =
+        jsx:decode(Data, [return_maps, {labels, atom}]),
+    Encoding = binary_to_atom(EncodingUtf8, utf8),
```

Comment:
I'm not sure if we can assume that it's an atom, according to http://erlang.org/doc/man/unicode.html#bom_to_encoding-1 the encoding can be a tuple. For context, I only looked into this because i was worried we might be generating arbitrary atoms but it looks like there is a finite number of them.